Reference Frame module abc2dq and dq2alphabeta

murgui opened the pull request:
It is missing to test with an actual sine as an input. I will do that after developing UART communication.

murgui committed:
Reference Frame module abc2dq and dq2alphabeta

AitorTeran commented on Software/Development/SemesterProject/App/Include/referenceFrameConversion.h:
```diff
@@ -0,0 +1,38 @@
+{
+    float DComponent;
+    float QComponent;
+}DQObject;
```
dComponent, qComponent and dqObject?¿

AitorTeran replied:
I see you called everything with capital letters, i think it's better with lower case but it's not a big deal so up to you

NicolaiFransen replied:
I actually agree with Aitor¿?¿

murgui replied:
Done

AitorTeran commented on Software/Development/SemesterProject/App/referenceFrameConversion.c:
```diff
@@ -0,0 +1,78 @@
+ *     abVoltage = dq2alphabeta(&DQVoltage, theta);
+ *
+ */
+alphaBetaObject dq2alphabeta(DQObject *DQVoltage, float thetaRad)
```
Why DQVoltage? shouldn't it be more general in the functio, dqObject?

murgui replied:
Done

AitorTeran reviewed (approved):
Looking good, ready to be shipped!

NicolaiFransen commented on Software/Development/SemesterProject/App/Include/referenceFrameConversion.h:
```diff
@@ -0,0 +1,38 @@
+{
+    float alphaComponent;
+    float betaComponent;
+}alphaBetaObject;
```
I think you should consider using afaBeta 🤔

murgui replied:
( ͡° ͜ʖ ͡°)

NicolaiFransen reviewed (commented):
Looks nice, but consider the capital letters

murgui committed:
Changes after review

NicolaiFransen reviewed (approved):
Ship it (y)

murgui committed:
Merged develop to get latest updates

murgui committed:
abc2dq and dq2afbe tested

In order to test, signalGenerator software component was developed and a pthon script for plotting UART data was developed, both uploaded with this commit. Also for determining the desired output, a PSIM simulation of the reference frame conversion was performed

murgui committed:
Merge branch 'develop' into feature/referenceFrameChange
